packages/vite/src/node/ssr/ssrExternal.ts

```diff
@@ -179,8 +179,21 @@ function collectExternals(
 
 export function shouldExternalizeForSSR(
   id: string,
-  externals: string[]
+  externals: string[],
+  noExternal: string | RegExp | (string | RegExp)[] | true
 ): boolean {
+  if (noExternal === true) {
+    return false
+  }
+  const noExternalsArr = Array.isArray(noExternal) ? noExternal : [noExternal]
+  for (const e of noExternalsArr) {
+    if (typeof e === 'string' && e === id) {
+      return false
+    }
+    if ((e as RegExp).test(id)) {
```

Comment:
**question:** Why is this ` as RegExp` needed?

Reply:
![image](https://user-images.githubusercontent.com/17424434/148488140-4ab4a2a9-fb73-414f-8718-b6f054b24125.png)

I don't know why there is not judge exactly

Reply:
Ah, it's because of the additional e === id check. If this is false e could still be a string. And then when calling test on a string it would be a bug.

Reply:
Yeah, has updated